backend/src/paymaster/index.ts

```diff
@@ -648,7 +649,7 @@ export class Paymaster {
       if (!oracleAggregator) {
         if (!isCoingeckoAvailable) throw new Error('Unable to fetch token price. Please try again later.')
         const {latestAnswer, decimals} = await this.getLatestAnswerAndDecimals(provider, nativeOracleAddress, chainId);
-        const data = await this.getPriceFromCoingecko(chainId, feeToken, latestAnswer, decimals);
+        const data = await this.getPriceFromCoingecko(chainId, feeToken, latestAnswer, decimals, log);
```

Comment:
If there is an error in `getPriceFromCoingecko` , this step will store junk. I want you to handle the error thrown from that function and retain the cached value as it is.